salvo/requirements.txt

```diff
@@ -1,35 +1,48 @@
 appdirs==1.4.4
-attrs==17.4.0
-boto3==1.16.22
-botocore==1.19.22
-certifi==2018.1.18
-chardet==3.0.4
-decorator==4.4.2
-distlib==0.3.1
-docker==2.5.1
-docker-pycreds==0.2.1
-filelock==3.0.12
-idna==2.6
-importlab==0.5.1
-importlib-metadata==1.7.0
-importlib-resources==3.0.0
-jmespath==0.10.0
-networkx==2.5
-ninja==1.10.0.post2
-pluggy==0.6.0
-py==1.10.0
-pybind11==2.6.1
-pygobject==3.26.1
-pytest==3.3.2
-python-apt==1.6.5+ubuntu0.3
-python-dateutil==2.8.1
-pytype==2020.11.12
-PyYAML==5.4
-requests==2.25.0
-s3transfer==0.3.3
-six==1.15.0
-typed-ast==1.4.1
-urllib3==1.26.5
-virtualenv==20.0.31
-websocket-client==0.44.0
-zipp==3.1.0
+attrs==21.4.0
```

Comment:
I know we did this before, however fixing requirements to a specific version (and on top of that to a specific patch version) will cause higher than usual amount of work in the future. I.e. there will be a large amount of changes we will have to do all at once.

What would it take to use `>=` instead of `==` ideally in all our dependencies? This will break CI when the dependency makes an incompatible change, but at least we will know about it soon. It should be part of our maintenance procedure to update the dependencies regularly.

Reply:
Thanks for this reminder, I have less exprience maintaining a python project, so I didn't notice this. 
It works when I use `>=` instead of `==` both in our dependencies in this PR or original dependencies. It will download the appropriate version automatically. So it's good enough to use '>=' in our original dependencies configuration , it works from python3.6 to python3.10. 

However, I can't understand this, can you give an example:
> This will break CI when the dependency makes an incompatible change, but at least we will know about it soon.

Reply:
Sorry, I should have provided more details. What I meant is - regardless of our configuration, the packages we use will evolve over time and eventually introduce a breaking change. We have two options and which we choose affects how and when we learn about the breaking change:

1. with `>=` our tests in AZP pick up the latest package with the breaking change and will report a test error. This gives us an early warning that we need to update our code due to the upstream breaking change.
2. with `==` our tests in AZP continue using the package version that is compatible with our code. Eventually we either decide to update the dependencies or we will be forced to due to a deprecation of one of the versions we use. This is the moment when we will discover all the breaking changes and since they accumulated - this could become a large and complex PR to get it all in order.

Personally I prefer (1), since it results in smaller (though multiple) updates that are easier to write and review.

Reply:
@mum4k Thanks for providing more details. I can understand completely now, and I also prefer (1) to avoid huge dependencies updates in the future. I will update this PR.